xarray/core/utils.py

```diff
@@ -470,27 +457,27 @@ class NDArrayMixin(NdimSizeLenMixin):
     `dtype`, `shape` and `__getitem__`.
     """
     @property
-    def dtype(self):
+    def dtype(self: Any) -> np.dtype:
         return self.array.dtype
 
     @property
-    def shape(self):
+    def shape(self: Any) -> Tuple[int]:
         return self.array.shape
 
-    def __getitem__(self, key):
+    def __getitem__(self: Any, key):
         return self.array[key]
 
-    def __repr__(self):
+    def __repr__(self: Any) -> str:
         return '%s(array=%r)' % (type(self).__name__, self.array)
 
 
-class ReprObject(object):
-    """Object that prints as the given value, for use with sentinel values."""
-
-    def __init__(self, value: str):
+class ReprObject:
+    """Object that prints as the given value, for use with sentinel values.
+    """
+    def __init__(self, value: str) -> None:
```

Comment:
note: I think mypy is OK with skipping `-> None` for `__init__` methods.

Reply:
I was going with PEP 484:
> (Note that the return type of __init__ ought to be annotated with -> None. The reason for this is subtle. If __init__ assumed a return annotation of -> None, would that mean that an argument-less, un-annotated __init__ method should still be type-checked? Rather than leaving this ambiguous or introducing an exception to the exception, we simply say that __init__ ought to have a return annotation; the default behavior is thus the same as for other methods.)

but I've now read https://github.com/python/mypy/issues/3358, which likely means a revision of the PEP in the opposite sense is incoming. I'll remove the ``-> None`` from ``__init__``.